api/core/v2/apikey.go

```diff
@@ -0,0 +1,70 @@
+package v2
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+)
+
+const (
+	// APIKeysResource is the name of this resource type.
+	APIKeysResource = "apikeys"
+)
+
+// StorePrefix returns the path prefix to this resource in the store.
+func (a *APIKey) StorePrefix() string {
+	return APIKeysResource
+}
+
+// URIPath returns the path component of an api key URI.
+func (a *APIKey) URIPath() string {
+	return path.Join(URLPrefix, APIKeysResource, url.PathEscape(a.Name))
+}
+
+// Validate returns an error if the CheckName and Subscription fields are not
+// provided.
+func (a *APIKey) Validate() error {
+	if a.Namespace != "" {
+		return fmt.Errorf("api key cannot have a namespace")
+	}
+
+	if a.Username == "" {
+		return fmt.Errorf("api key must have a username")
+	}
```

Comment:
So I'm a bit confused since the specs are no longer in sync with what we appear to be implementing, but I'm wondering if we should validate whether the user exist in the Sensu store; I assume we only want to support API keys for "internal" users, and not users authenticated via an authentication provider, since we can't retrieve their groups on-demand.

Reply:
We would probably validate this in the API router. Not sure why this simple validation isn't in sync with the spec. Could you elaborate?

Reply:
I've added user validation to the router.